org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/KotlinDefaultArgumentsFilter.java

```diff
@@ -132,15 +138,21 @@ public void match(final MethodNode methodNode,
 
 		private static int maskVar(final String desc,
 				final boolean constructor) {
+			final Type[] argumentTypes = Type.getMethodType(desc)
+					.getArgumentTypes();
+			int masks;
+			for (masks = 1; masks <= 8; masks++) {
+				if (argumentTypes.length <= masks * 32 + masks + 1) {
+					break;
+				}
+			}
```

Comment:
Mathematically equivalent to:

```suggestion
			int masks = (argumentTypes.length - 2) / 33 + 1;
```

Extract this to its own method with documentation and unit test?

Reply:
oh no, increase of method visibility for testing 🙀 😉